src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,72 @@
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { Filter } from '../../types/Filter';
+import { FilterParam } from '../../App';
+
+type Props = {
+  filter: string;
+  setFilter: (a: Filter) => void;
+  todos: Todo[];
+  handleDeleteTodo: (a: number) => void;
+  setLoadingIds: (a: number[]) => void;
+};
+
+export const Footer: React.FC<Props> = ({
+  filter,
+  setFilter,
+  todos,
+  handleDeleteTodo,
+  setLoadingIds,
+}) => {
+  const handleFilter = (event: React.MouseEvent<HTMLElement>) => {
+    const filterValue = event.currentTarget.textContent as Filter;
+
+    setFilter(filterValue);
+  };
+
+  const handleClearComplete = () => {
+    const completedTodo = todos.filter(todo => todo.completed);
+
+    Promise.allSettled(completedTodo).then(() =>
+      completedTodo.map(todo => {
+        setLoadingIds(current => [...current, todo.id]);
```

Comment:
loader should be settled befor request and removed after response.
allSettled should accept an array of promises.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled
tests expects, that todos, which was failed during remove on server not be deleted on client.

![Screenshot 2024-12-04 at 14 57 53](https://github.com/user-attachments/assets/84096223-8c20-4619-84c4-3986ce468e66)